src/libraries/System.Private.CoreLib/src/System/SearchValues/Strings/StringSearchValues.cs

```diff
@@ -309,25 +309,16 @@ static SearchValues<string> PickAhoCorasickImplementation<TCaseSensitivity>(AhoC
 
             if (Vector128.IsHardwareAccelerated && value.Length > 1 && value.Length <= maxLength)
             {
-                if (!ignoreCase)
+                SearchValues<string>? searchValues = value.Length switch
                 {
-                    return new SingleStringSearchValuesThreeChars<CaseSensitive>(uniqueValues, value);
-                }
-
-                if (asciiLettersOnly)
-                {
-                    return new SingleStringSearchValuesThreeChars<CaseInsensitiveAsciiLetters>(uniqueValues, value);
-                }
+                    < 4 => TryCreateSingleValuesThreeChars<ValueLengthLessThan4>(value, uniqueValues, ignoreCase, allAscii, asciiLettersOnly),
+                    < 8 => TryCreateSingleValuesThreeChars<ValueLength4To7>(value, uniqueValues, ignoreCase, allAscii, asciiLettersOnly),
+                    _ => TryCreateSingleValuesThreeChars<ValueLength8OrLongerOrUnknown>(value, uniqueValues, ignoreCase, allAscii, asciiLettersOnly),
```

Comment:
Where does the "OrUnknown" come into play?

Reply:
In this case `Unknown` is used when this shared logic is called by Teddy to confirm matches.
I just opted to merge the `>= 8` and `Unknown` since the handling logic for both happens to be the same right now.

Reply:
Ok